birdhouse/pavics-compose.sh

```diff
@@ -125,19 +125,11 @@ if [[ $1 == "up" ]]; then
 
   # no error if already exist
   # create externally so nothing will delete these data volume automatically
-  docker volume create ${DOCKER_JUPYTERHUB_USER_PERSISTENCE_VOLUME}  # users notebooks
   docker volume create jupyterhub_data_persistence  # jupyterhub db and cookie secret
   docker volume create thredds_persistence  # logs, cache
 
-  # set proper write permissions for image pavics/workflow-tests
-  # this is shared for all users, no multi users isolation
-  # see open issue https://github.com/moby/moby/issues/32582 [feature] Allow
-  # mounting sub-directories of named volumes
-  # for multi user isolation, we should leverage sub-directories of named
-  # volumes
-  docker run --rm --name set_permissions \
-    -v ${DOCKER_JUPYTERHUB_USER_PERSISTENCE_VOLUME}:/notebook_dir bash \
-    bash -c "chown 1000:1000 /notebook_dir"
+  # Create the path to contain user notebooks if it doesn't exist
+  mkdir -p "${JUPYTERHUB_USER_DATA_DIR}"
```

Comment:
Ah !  The base user data is created here !  This will likely not work since the base `/data` is most likely owned by `root` while the user running this `pavics-compose.sh` script must not be `root` else we run into weirdo problems (Dominique Lenoir found out about this the hard way previously).

You'll probably need the `docker run --rm --name set_permissions` trick below to create this base dir.

Reply:
I... think it's not even necessary to create it here. It's always mounted by docker, which will create it as root if it doesn't exist. Do you think of a use case where this folder is not mounted by docker?

Reply:
Agreed, probably not needed to be created since docker will create itself on demand.